src/Battlescape/ActionMenuState.cpp

```diff
@@ -465,6 +523,68 @@ void ActionMenuState::btnActionMenuItemClick(Action *action)
 			}
 			_game->popState();
 		}
+		else if (weapon->getBattleType() == BT_SCRIPTED)
+		{
+			// check TUs first, then execute action
+			// NOTE: Fine tuning needed for melee, etc.
+			if (!_action->haveTU(&_action->result))
+			{
+				//nothing
+			}
+			else
+			{
+				TileEngine *tileEngine = _game->getSavedGame()->getSavedBattle()->getTileEngine();
+				// is targeting action? (set _action->targeting = true, newHitLog = true)
+				// is melee action? newHitLog = true
+				// action response to the user? (set _action->result)
+				// is simple activation?
+				tileEngine->scriptedItemUse(_action);
+				
+				// action could have been changed by script now
+				switch (_action->type) {
+					case BA_PRIME: // default prime code
```

Comment:
BA_UNPRIME missing or handled by the default clause?

Reply:
Should be handled by default clause, iirc.